src/lib.rs

```diff
@@ -933,58 +934,39 @@ pub fn weak_rng() -> XorShiftRng {
     XorShiftRng::new().unwrap()
 }
 
-/// Controls how the thread-local RNG is reseeded.
-#[cfg(feature="std")]
-#[derive(Debug)]
-struct ThreadRngReseeder;
```

Comment:
Unnecessary with `ReseedWithNew`.
Also I moved around the rest op the `ThreadRng` code a bit, partly to make the copy easier, partly a little cleanup.